cmd/collector/app/handler/grpc_handler.go

```diff
@@ -98,12 +99,22 @@ func (c *batchConsumer) validateTenant(ctx context.Context) (string, error) {
 		return "", nil
 	}
 
+	httpMd := client.FromContext(ctx)
+	tenants := httpMd.Metadata.Get(c.tenancyMgr.Header)
+
+	if len(tenants) > 0 {
+		if !c.tenancyMgr.Valid(tenants[0]) {
```

Comment:
We don't want to duplicate validation. You are missing other checks like the one that causes `return "", status.Errorf(codes.PermissionDenied, "missing tenant header")`

This function needs to be constructed of two parts, one is extracting the tenant info, another is validation.

And I would prefer to consolidate all this in pkg/tenancy/getValidTenant (which can be made public). It's a kind of a mess there too. We have four possible sources of tenant:
  * it may already be in the Context
  * it may be in the HTTP request headers
  * it may be in gRPC request headers, which are accessible via Context via `metadata`
  * finally, it may be in the Context as OTEL's `client.Metadata`